File: easy/393_MakingChange.r
change <- function(x, denom = c(500, 100, 25, 10, 5, 1)){
    if(denom[1] == 1){return(x)}
    else{
        return(floor(x / denom[1]) + change(x %% denom[1], denom[2:length(denom)]))
    }
}

File: easy/393_MakingChange.c
#include <stdio.h>

int denom[] = {500, 100, 25, 10, 5, 1};

int change(int x, int *c){
    int t = c[0];
    if(t == 1) return x;
    else return x / t + change(x % t, &c[1]);
}

int main(int argc, char **argv){
    printf("change(0) => %d\n", change(0, denom));
    printf("change(12) => %d\n", change(12, denom));
    printf("change(468) => %d\n", change(468, denom));
    printf("change(123456) => %d\n", change(123456, denom));
}
